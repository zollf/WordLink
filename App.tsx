import React, { useEffect, useMemo, useState } from 'react';
import { ImageBackground } from 'react-native';
import { Raleway_400Regular, Raleway_700Bold, useFonts } from '@expo-google-fonts/raleway';
import { configure } from 'mobx';
import { Provider, observer } from 'mobx-react';

import { gameStore, global } from './stores';
import _getData from './lib/getData';
import { Game, Index, Menu, Profile, Welcome, Settings } from './src';
import image from './images/background.png';

import styles from './styles/main';

configure({ enforceActions: 'observed' });

const App = () => {
  const [, setData] = useState({});

  const [fontsLoaded] = useFonts({
    Raleway_400Regular,
    Raleway_700Bold,
  });

  useEffect(() => {
    _getData().then((v) => {
      if (v.success) {
        setData(v.value);
      }
    });
  }, []);

  const loadPage = useMemo(() => {
    const pages: Pages = {
      index: <Index />,
      menu: <Menu />,
      profile: <Profile />,
      game: <Game />,
      settings: <Settings />,
    };

    if (global.onStartingPage) {
      return <Welcome />;
    } else {
      return pages[global.currentPage];
    }
  }, [global.onStartingPage, global.currentPage]);

  return (
    <Provider global={global} gameStore={gameStore}>
      <ImageBackground source={image} style={styles.image}>
        {fontsLoaded && loadPage}
      </ImageBackground>
    </Provider>
  );
};

export default observer(App);
