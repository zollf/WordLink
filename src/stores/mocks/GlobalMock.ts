const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
