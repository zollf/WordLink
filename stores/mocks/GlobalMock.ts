const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
