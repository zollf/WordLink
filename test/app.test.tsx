import React from 'react';
//@ts-ignore there are two app file names [json, tsx], todo: work-around 
import App from '../App.tsx';
import { Provider } from 'mobx-react';
import GlobalMock from '../stores/mocks/GlobalMock';

import renderer from 'react-test-renderer';

const component = (
  <Provider global={GlobalMock}>
    <App />
  </Provider>
);

describe('<App/>', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});