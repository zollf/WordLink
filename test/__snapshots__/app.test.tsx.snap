// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> matches snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../images/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      data-test-id="wrapper"
      style={
        Object {
          "alignItems": "center",
          "bottom": 80,
          "display": "flex",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        data-test-id="menu-bar"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C8C8C8",
            "borderRadius": 5,
            "display": "flex",
            "flexDirection": "row",
            "height": 75,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 212,
            "zIndex": -1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "display": "flex",
              "height": 50,
              "justifyContent": "center",
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 0.1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../images/home.png",
              }
            }
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "display": "flex",
              "height": 50,
              "justifyContent": "center",
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../images/gamepad.png",
              }
            }
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "display": "flex",
              "height": 50,
              "justifyContent": "center",
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 0.1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../images/avatar.png",
              }
            }
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "borderRadius": 5,
              "height": 50,
              "left": 80.5,
              "position": "absolute",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
              "zIndex": -1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
